extension/src/patch.ts

```diff
@@ -0,0 +1,187 @@
+  name: string,
+  expData: ExperimentsOutput
+): ExperimentDetails | undefined => {
+  for (const [sha, experimentsObject] of Object.entries(
+    omit(expData, EXPERIMENT_WORKSPACE_ID)
+  )) {
+    const details = findExperimentByName(name, sha, experimentsObject)
+    if (details) {
+      return details
+    }
+  }
+}
+
+const sendPostRequest = (
+  studioAccessToken: string,
+  body: StartRequestBody | DataRequestBody | DoneRequestBody
+): Promise<Response> =>
+  fetch(STUDIO_ENDPOINT, {
+    body: JSON.stringify(body),
+    headers: {
+      Authorization: `token ${studioAccessToken}`,
+      'Content-type': 'application/json'
+    },
+    method: 'POST'
+  })
+
+const shareWithProgress = (
+  experimentDetails: ExperimentDetails,
+  repoUrl: string,
+  studioAccessToken: string
+): Thenable<unknown> =>
+  Toast.showProgress('Sharing Experiment', async progress => {
+    const { metrics, params, baselineSha, name } = experimentDetails
+    const base: BaseRequestBody = {
+      baseline_sha: baselineSha,
+      client: 'vscode',
+      name,
+      repo_url: repoUrl
+    }
+
+    progress.report({
+      increment: 0,
+      message: 'Initializing experiment...'
+    })
+    await sendPostRequest(studioAccessToken, { ...base, type: 'start' })
+
+    progress.report({ increment: 33, message: 'Sending data...' })
+    await sendPostRequest(studioAccessToken, {
+      ...base,
+      metrics: metrics || {},
+      params: params || {},
+      plots: {},
+      step: 0,
+      type: 'data'
+    })
+
+    progress.report({ increment: 33, message: 'Completing process...' })
+    await sendPostRequest(studioAccessToken, { ...base, type: 'done' })
+
+    progress.report({ increment: 33, message: 'Done' })
+
+    return Toast.delayProgressClosing()
+  })
+
+const pushExperiment = async (
+  internalCommands: InternalCommands,
+  dvcRoot: string,
+  name: string,
+  studioAccessToken: string
+) => {
+  const [repoUrl, expData] = await Promise.all([
+    internalCommands.executeCommand(
+      AvailableCommands.GIT_GET_REMOTE_URL,
+      dvcRoot
+    ),
+    internalCommands.executeCommand<ExperimentsOutput>(
+      AvailableCommands.EXP_SHOW,
+      dvcRoot,
+      ExperimentFlag.NO_FETCH
+    )
+  ])
+
+  const experimentDetails = collectExperimentDetails(name, expData)
+
+  if (!repoUrl) {
+    return Toast.showError(
+      'Failed to share experiment, unable to generate Git repo URL'
+    )
+  }
+
+  if (!experimentDetails) {
+    return Toast.showError(
+      'Failed to share experiment, unable to locate the required data'
+    )
+  }
+
+  return shareWithProgress(experimentDetails, repoUrl, studioAccessToken)
+}
+
+export const registerPatchCommand = (internalCommands: InternalCommands) =>
+  internalCommands.registerCommand(
+    AvailableCommands.EXP_PUSH,
+    (...args: Args) => {
+      const [studioAccessToken, dvcRoot, name] = args
+
+      return pushExperiment(internalCommands, dvcRoot, name, studioAccessToken)
```

Comment:
[F] The way that I've split up the functionality/built this feature means that this part can be replaced by a CLI call once it catches up 🤞🏻.

Reply:
The above is the main reason that (for now at least) I have not made an effort to reuse existing data/fit in more with the rest of the codebase.